src/Action/SetObjectFieldValueAction.php

```diff
@@ -40,18 +42,40 @@ final class SetObjectFieldValueAction
      */
     private $validator;
 
-    public function __construct(Environment $twig, Pool $pool, $validator)
+    /**
+     * @var DataTransformerResolver
+     */
+    private $resolver;
+
+    /**
+     * @param ValidatorInterface           $validator
+     * @param DataTransformerResolver|null $resolver
+     */
+    public function __construct(Environment $twig, Pool $pool, $validator, $resolver = null)
     {
         // NEXT_MAJOR: Move ValidatorInterface check to method signature
         if (!($validator instanceof ValidatorInterface)) {
             throw new \InvalidArgumentException(
-                'Argument 3 is an instance of '.\get_class($validator).', expecting an instance of'
-                .' \Symfony\Component\Validator\Validator\ValidatorInterface'
+                'Argument 3 is an instance of '.\get_class($validator).', expecting an instance of '
+                .ValidatorInterface::class
             );
         }
+
+        // NEXT_MAJOR: Move DataTransformerResolver check to method signature
+        if (!($resolver instanceof DataTransformerResolver)) {
+            @trigger_error(sprintf(
+                'Passing other type than %s in argument 4 to %s() is deprecated since sonta-project/admin-bundle 3.x and will throw %s exception in 4.0.',
```

Comment:
```suggestion
                'Passing other type than %s in argument 4 to %s() is deprecated since sonata-project/admin-bundle 3.x and will throw %s exception in 4.0.',
```